### PR TITLE
Update and cross-reference migration notes.

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

Migration notes were a bit old and not cross-referenced to the rest of the documentation

### Proposed changes

- juice up migration notes

 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

----
📚 Documentation preview 📚: https://datacube-core--1686.org.readthedocs.build/en/1686/

### CODE REVIEW

File: docs/installation/MIGRATION-1.8-to-1.9.rst [hunk 1]
```diff
@@ -123,13 +137,19 @@ Major Changes between 1.8.x and 1.9.x
...
    the new lineage API.
 
-6. Support for multi-dimensional loading of hyperspectral datasets (Coming Soon)
+6. Support for multi-dimensional loading of hyperspectral datasets
 
-   This is a work in progress and will not be available in 1.9.0. It will appear in a later 1.9.x release.
+   Use of this feature currently requires installing `odc-stac`_.
+
+   If `odc-stac`_ is installed, you may specify `driver='rio'` to access the new hyperspectral loading and
+   dask-chunking features of ``stac_load``.
```
Comment (omad) [inline docs/installation/MIGRATION-1.8-to-1.9.rst]:
```suggestion
   Use of this feature currently requires installing `odc-loader`_.

   If `odc-loader`_ is installed, you may specify `driver='rio'` or `driver='zarr'` when loading to access the new hyperspectral loading and
   dask-chunking features.
```

I probably should add an installation extra too...

File: docs/installation/MIGRATION-1.8-to-1.9.rst [hunk 2]
```diff
@@ -123,13 +137,19 @@ Major Changes between 1.8.x and 1.9.x
...
 
 7. The long-deprecated "ingestion" workflow and "executor" API have both been removed.
 
-8. Multiple locations per dataset is now deprecated.
+8. Multiple locations per dataset is now deprecated, and is not supported by the ``postgis`` index driver.
+
+.. _`odc-geo`: https://github.com/opendatacube/odc-geo
+.. _`odc-stac`: https://github.com/opendatacube/odc-stac
```
Comment (omad) [inline docs/installation/MIGRATION-1.8-to-1.9.rst]:
```suggestion
.. _`odc-loader`: https://github.com/opendatacube/odc-loader
```

**Review by omad — APPROVED** [review]
Nice, the more interlinking the better IMO. :)

**Comment by SpacemanPaul** [comment]
> Nice, the more interlinking the better IMO. :)

Yeah, me too.  Sphinx seems to always work in not quite the manner you expect though.

**Pull request merged**